File: app/src/main/java/org/intelehealth/ekalarogya/utilities/ResponseChecker.java
package org.intelehealth.ekalarogya.utilities;

import retrofit2.Response;

public class ResponseChecker<T> {
    private final Response<T> response;

    public ResponseChecker(Response<T> response) {
        this.response = response;
    }

    public boolean isNotAuthorized() {
        return response.code() == 401;
    }
}

File: videolibrary/src/main/java/org/intelehealth/videolibrary/utils/ResponseChecker.kt
package org.intelehealth.videolibrary.utils

import retrofit2.Response

/**
 * Created by Arpan Sircar. on 08-02-2024.
 * Email : arpan@intelehealth.org
 * Mob   : +919123116015
 **/

class ResponseChecker<T>(private val response: Response<T>) {
    val isNotAuthorized: Boolean
        get() = response.code() == 401
}